launcher/ui/dialogs/BlockedModsDialog.cpp

```diff
@@ -1,28 +1,191 @@
+    watcher.addPath(downloadsFolder);
+    watcher.addPath(modsFolder);
+}
+
+
+/// @brief scan all watched folder
+void BlockedModsDialog::scanPaths() {
+    for (auto &dir : watcher.directories()) {
+        scanPath(dir);
+    }
+}
+
+/// @brief Scan the directory at path, skip paths that do not contain a file name 
+///        of a blocked mod we are looking for
+/// @param path the directory to scan
+void BlockedModsDialog::scanPath(QString path) {
+
+    QDir scan_dir(path);
+    QDirIterator scan_it(path, QDir::Filter::Files | QDir::Filter::Hidden, QDirIterator::NoIteratorFlags);
+    while (scan_it.hasNext()) {
+        QString file = scan_it.next();
+
+        if (!checkValidPath(file)) {
+            continue;
+        }
+
+        auto hash_task = Hashing::createBlockedModHasher(file, ModPlatform::Provider::FLAME, "sha1");
+
+        qDebug() << "Creating Hash task for path: " << file;
+
+        connect(hash_task.get(), &Task::succeeded, [this, hash_task, file] { 
+            checkMatchHash(hash_task->getResult(), file);
+        });
+        connect(hash_task.get(), &Task::failed, [this, hash_task, file] { 
+            qDebug() << "Failed to hash path: " << file;
+        });
+        
+        hashing_task->addTask(hash_task);
+    }
+
+    hashing_task->start();
+
+}
+
+/// @brief check if the computed hash for the provided path matches a blocked
+///        mod we are looking for
+/// @param hash the computed hash for the provided path
+/// @param path the path to the local file being compared
+void BlockedModsDialog::checkMatchHash(QString hash, QString path) {
+    bool match = false;
+
+    qDebug() << "Checking for match on hash: " << hash << " | From path:" << path;
+
+    for (auto &mod : mods) {
+        if (mod.matched) {
+            continue;
+        }
+        if (mod.hash.compare(hash, Qt::CaseInsensitive) == 0) {
+            mod.matched = true;
+            mod.localPath = path;
+            match = true;
+
+            qDebug() << "Hash match found: " << mod.name << " " << hash << " | From path:" << path;
+
+            break;
+        }
+    }
+
+    if (match) {
+        update();
     }
 }
+
+/// @brief Check if the name of the file at path matches the name of a blocked mod we are searching for
+/// @param path the path to check
+/// @return boolean: did the path match the name of a blocked mod?
+bool BlockedModsDialog::checkValidPath(QString path) {
+
+    QFileInfo file = QFileInfo(path);
+    QString filename = file.fileName();
+
+    for (auto &mod : mods) {
+        if (mod.name.compare(filename, Qt::CaseInsensitive) == 0) {
+            qDebug() << "Name match found: " << mod.name << " | From path:" << path;
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool BlockedModsDialog::allModsMatched() {
+    for (auto &mod : mods) {
+        if (!mod.matched)
+            return false;
+    }
+    return true;
```

Comment:
You can use `std::all_of` if you want ^^
https://en.cppreference.com/w/cpp/algorithm/all_any_none_of

Reply:
ty

Reply:
no problem :)